src/commit.ts

```diff
@@ -58,13 +72,18 @@ function processCommits(commitHeaders: string[]): ICommitData {
       }
   
       const hash = words[0];
-      const prefix = words[1].split(':')[0];
+      const { type, scope } = getTypeAndScope(words[1]);
       const rest = words.slice(2).join(' ');
   
-      const lineDescription = `${rest} (${hash})`;
-      if (prefix in processedCommits) {
-        console.log(`adding "${lineDescription}" to ${prefix}`);
-        processedCommits[prefix].push(lineDescription);
+      const lineDescription = [
+        scope ? `${scope}:` : undefined,
+        rest,
+        `(${hash})`
+      ].filter(part => !!part).join(' ');
```

Comment:
I've split these sections up as scope is optional and a ternary inside a template literal is confusing.